src/pointings_search/geometry.py

```diff
@@ -0,0 +1,80 @@
+"""Helper functions for basic geometric operations."""
+
+import numpy as np
+
+
+def ang2unitvec(ra, dec):
+    """Convert an angular direction (RA, dec) to a unit vector (x, y, z).
+
+    Parameters
+    ----------
+    ra : `float` or numpy array
+       The right ascension in degrees.
+    dec : `float` or numpy array
+        The declination in degrees.
+
+    Returns
+    -------
+    (x, y, z) : `tuple`
+        The unit vector as a tuple.
+    """
+    ra_rad: float = ra * (np.pi / 180.0)
+    dec_rad: float = dec * (np.pi / 180.0)
+
+    x = np.cos(ra_rad) * np.cos(dec_rad)
+    y = np.sin(ra_rad) * np.cos(dec_rad)
+    z = np.sin(dec_rad)
+
+    return (x, y, z)
+
+
+def unitvec2ang(x, y, z):
+    """Convert a unit vector (x, y, z) to an angular direction (RA, dec).
+
+    Parameters
+    ----------
+    x : `float` or numpy array
+    y : `float` or numpy array
+    z : `float` or numpy array
+
+    Returns
+    -------
+    ra : `float` or numpy array
+       The right ascension(s) in degrees.
+    dec : `float` or numpy array
+        The declination(s) in degrees.
+    """
+    ra = (180.0 / np.pi) * np.arctan2(y, x)
+    ra = ra % 360.0  # Only positive values
+    dec = (180.0 / np.pi) * np.arcsin(z)
+    return ra, dec
+
+
+def angular_distance(pts1, pts2):
```

Comment:
this is the formula for angular distance between two normal vectors right? we should specify that so nobody tries shove ra/dec in there 😭 

also generally I would default to using astropy's distance calculation because they cover a lot of special cases, specifically at the poles, although there's is very slow so I understand not wanting to use it. and also the ecliptic doesn't pass through the poles afaik. do we have a sense of scale for the amount of points that we plan on running through this?